packages/flutter/lib/src/material/scaffold.dart

```diff
@@ -24,6 +24,21 @@ const double _kFloatingActionButtonMargin = 16.0; // TODO(hmuller): should be de
 const Duration _kFloatingActionButtonSegue = const Duration(milliseconds: 200);
 final Tween<double> _kFloatingActionButtonTurnTween = new Tween<double>(begin: -0.125, end: 0.0);
 
+/// Returns a path for a notch in the outline of a shape.
+///
+/// The path makes a notch in the host shape that can contain the guest shape.
+///
+///
+/// The `host` is the bounding rectangle for the shape into which the notch will
+/// be applied.
+///
+///
+/// The `start` and `end` arguments are points on the outline of the host shape
+/// that will be connected by the returned path.
+///
+/// The returned path might pass inside the content bounds area.
```

Comment:
this should describe the contract better. For example, can the path contain multiple subpaths? Can it end with a call to close()? How will it be used?

(If you require that it's one subpath and that it doesn't call close, consider using extendWithPath instead, and saying that subsequent subpaths are ignored.)

Reply:
more contract questions: can the Path be mutated after it's returned? Can the path go outside the `guest` rectangle? outside the `host` rectangle? What order will `start` and `end` be in terms of the geometry? Will they always have the same `y` coordinate; will start always be to the left of end, etc.

Reply:
I added some docs.
The only guarantee for the return path is that it starts at `start` and ends at `end` and does not end with close(), anything else is allowed. I mentioned that, I emphasized that it can go inside the `guest` bounds, and that it can have multiple subpaths, as I think these are the things that might not be obvious. But I feel it is over verbose to enumerate a list of things the path is allowed to do...

Specifically the notchmaker used for the FAB can make further assumptions about the input. I added these assumptions to the documentation of `setFloatingActionButtonMakerFor`.

Regarding being able to mutate the path after it's returned, the answer is yes, but I do not see that we mention it in documentation for methods that returns a Path (e.g ShapeBorder.getOuterPath). I would expect the documentation to mention it only if the Path cannot be mutated.